Update spec parameters with ApiKeyAuth security scheme

svetlana-v opened the pull request:
Fixes:

Fixes bad request error api key not in spec when the ApiKeyAuth security method is used and the key parameter is sent in a query.
See https://github.com/zalando/connexion/issues/470.


Changes proposed in this pull request:

 - Update the spec parameters with the ApiKeyAuth security scheme if the corresponding security method is used

RobbeSneyders commented on connexion/operations/openapi.py:
```diff
@@ -335,3 +339,10 @@ def _get_val_from_param(self, value, query_defn):
         else:
             return make_type(value, query_schema["type"])
+
+    def _get_apikey_parameter(self, app_security, security_schemes):
```
Seems like this method can be shared between swagger2 and openapi3?

svetlana-v replied:
Yes, it's exactly the same. Was not sure where it can fit though. I don't really think it should be in their `utils` and also a bit weird to create a new helper file for it.

I also see now that it at least should be static here.

RobbeSneyders replied:
Why not in the abstract base class?

RobbeSneyders commented on connexion/operations/openapi.py:
```diff
@@ -335,3 +339,10 @@ def _get_val_from_param(self, value, query_defn):
             return make_type(value, query_schema["type"])
+
+    def _get_apikey_parameter(self, app_security, security_schemes):
+        security_methods = set().union(*(s.keys() for s in app_security))
```
You can simplify this to:
`security_methods = set().union(*app_security)`

svetlana-v replied:
Thanks!

RobbeSneyders reviewed (commented):
Looks a lot cleaner like this :+1: Left some additional comments.

drewdogg committed:
Fix unit tests on Windows (#1071)

aijanai committed:
Fixed mocking function not working with examples under numeric status codes (e.g., 200) , due to status code string being casted as integer and not raising a ValueError. (#1088)

Included tests that break without the patch (it was testing for 'default' only, but not for numeric codes like '200')

terrycorley committed:
Fix return type docstring for get_response (#912)

pbasista committed:
Do not sanitize body keys in OpenAPI 3 (#1008)

* Remove the unused "query_sanitazion" fixture

* Test whether no sanitization is performed in the request body

* Do not perform sanitization on request body keys in OpenAPI v3

The deserialized JSON form of the request body
needs to be passed to the client applications
* without further modification *
so that they can work directly with objects
that have been received over the network.
The only names for which sanitization makes sense
are the ones which are used as Python identifiers.

Keys of the top-level JSON object within the request payload
are never used by Connexion as Python identifiers.

Also, no such sanitization of keys within request body
is performed in OpenAPI v2.

Closes issue #835.

ddurham2 committed:
[#935] supporting application/json within multipart/form-data request body (#980)

* [#935] adding support to automatically decode json when a multipart/form-data contains a field with an application/json content-type

* Update connexion/decorators/uri_parsing.py

Co-Authored-By: Tomasz Trębski <kornicameister@gmail.com>

* Update connexion/decorators/uri_parsing.py

accepting other non-standard json content types too

Co-Authored-By: Tomasz Trębski <kornicameister@gmail.com>

* Update uri_parsing.py

syntax fix

* syntax fix

* fixes

* Adding unit test to verify that multipart/form-data values which are json objects are decoded and validated as such

Commit:
Datetime serialization (#851)

* Add datetime and uuid serialization for AioHttp

* Remove ujson dependency

* fix merge error

* Retry CI

* remove bad jsonifier import

* remove ujson import

Commit:
Updated API Key example link to correct location (#1089)

* Updated API Key example link to correct location

* Fixed example href location per PR feedback

cedricmessiant committed:
Allow users to enable debug mode via environment variables (#1084)

Commit:
fixed mocks not honouring OpeanAPI 3 examples standard (printing "value" key in output examples) (#1091)

Commit:
Pop apiKey from query parameters in security decorator to fix #470
